src/routes/snippets.rs

```diff
@@ -0,0 +1,55 @@
+use rocket::State;
+use rocket_contrib::json::{Json, JsonValue};
+use serde::Deserialize;
+
+use crate::application::Config;
+use crate::errors::ApiError;
+use crate::storage::{Changeset, Snippet, Storage};
+
+#[derive(Deserialize)]
+pub struct NewSnippet {
+    /// Snippet title. May be omitted in the user request.
+    pub title: Option<String>,
+    /// Snippet syntax. May be omitted in the user request.
+    pub syntax: Option<String>,
+    /// Snippet content. Must be specified in the user request. Can't be empty.
+    pub content: String,
+    /// List of tags attached to the snippet. May be omitted in the user
+    /// request.
+    pub tags: Option<Vec<String>>,
+}
+
+#[post("/snippets", data = "<body>")]
+pub fn create_snippet(
+    config: State<Config>,
+    storage: State<Box<dyn Storage>>,
+    body: Json<NewSnippet>,
+) -> Result<JsonValue, ApiError> {
+    let NewSnippet {
+        title,
+        syntax,
+        content,
+        tags,
+    } = body.0;
+
+    if content.is_empty() {
+        return Err(ApiError::BadRequest(String::from(
+            "`content` - empty values not allowed.",
```

Comment:
nit: Should we use `:` instead of `-`? Python usually uses the former in similar situations.

Reply:
Actually, I copied the error message from the test we had in branch v4 :) https://github.com/xsnippet/xsnippet-api/blob/version/4/tests/resources/test_snippets.py#L277-L300

I don't have a strong preference, though.

Reply:
It was a `nit`. I think we can think of better error messages later on.

Reply:
Yeah, sounds good! I believe this one came from the validation library that we used in v4. I don't think the uses of the API should rely on the exact content of the error messages, so changing it is not a big deal.